mlir/utils/jenkins/miopen-tests/miopen_validate.sh

```diff
@@ -59,13 +66,36 @@ function parse_options() {
     done
 
     # Check required args
-    [[ $got_layout == 1 && $got_direction == 1 && $got_dtype == 1 ]] || usage
+    [[ $got_layout == 1 && $got_direction == 1 && $got_dtype == 1 ]] || [[ $TESTALL == 1 ]] || usage
 
     # Validate options
-    case "$DTYPE" in
-        fp16 | fp32 ) ;;
-        * ) echo "$0: Invalid dtype $DTYPE"; usage ;;
-    esac
+    if [[ $got_dtype == 1 ]]; then
+        case "$DTYPE" in
+            fp16 | fp32 ) ;;
+            * ) echo "$0: Invalid dtype $DTYPE"; usage ;;
+        esac
+    fi
+
+    # Detect XDLOPS if not specified
+    local xdlops
+    if [[ $XDLOPS == -1 ]]; then
+      xdlops=$(/opt/rocm/bin/rocm_agent_enumerator |grep gfx908)
```

Comment:
My brain is saying `grep -q` and checking the exit code in the if statement

That is dropping `xdlops` and going
```
if /opt/rocm/bin/rocm_agent_enumerator | grep -q gfx908; then
...
```

but that's a really tiny nitpick